src/layouts/QuickStartLayout.js

```diff
@@ -8,19 +8,15 @@ import PropTypes from 'prop-types';
 import { css } from '@emotion/react';
 import '../components/styles.scss';
 
-const getSidebarWidth = () => {
-  switch (true) {
-    case !window.location:
-      return 0;
-    case window.location.pathname === '/instant-observability/':
-      // this value matches '--sidebar-width' variable defined in instant-observability.js
-      return 300;
-    default:
-      return 0;
-  }
-};
-
 const QuickStartLayout = ({ children }) => {
+  const [sidebarWidth, setSidebarWidth] = useState(0);
+
+  useEffect(() => {
+    if (window.location.pathname === '/instant-observability/') {
+      setSidebarWidth(300);
+    } else setSidebarWidth(0);
+  }, [children]);
```

Comment:
Out of curiosity, do we need to add `children` to the dependency array? This might cause unnecessary calls to this function (whenever anything inside this component changes, this will get called).

Reply:
because this is in the layout it won't fire the effect again (it didn't seem) when switching to the details view since the layout doesn't change. using the children of the layout to determine if it fires was the only way i found to re-trigger it. very open to other solutions though! i just haven't found them 😭